test/extended/monitoring/collection_profiles.go

```diff
@@ -0,0 +1,219 @@
+	"k8s.io/client-go/kubernetes"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	pov1api "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	pov1client "github.com/prometheus-operator/prometheus-operator/pkg/client/versioned/typed/monitoring/v1"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+const (
+	projectName = "monitoring-collection-profiles"
+
+	collectionProfileFeatureLabel = "monitoring.openshift.io/collection-profile"
+	collectionProfileFull         = "full"
+	collectionProfileDefault      = collectionProfileFull
+	collectionProfileMinimal      = "minimal"
+
+	operatorNamespaceName     = "openshift-monitoring"
+	operatorConfigurationName = "cluster-monitoring-config"
+)
+
+var (
+	oc   = exutil.NewCLI(projectName)
+	tctx = context.Background()
+
+	collectionProfilesSupportedList = []string{
+		collectionProfileFull,
+		collectionProfileMinimal,
+	}
+)
+
+type runner struct {
+	kclient kubernetes.Interface
+	mclient *pov1client.MonitoringV1Client
+}
+
+var _ = g.Describe("[sig-instrumentation][OCPFeatureGate:MetricsCollectionProfiles] All collection profiles should have at-least one implementation", func() {
+	defer g.GinkgoRecover()
+
+	o.SetDefaultEventuallyTimeout(5 * time.Minute)
+	o.SetDefaultEventuallyPollingInterval(5 * time.Second)
+
+	var err error
+	r := &runner{}
+	r.kclient, err = kubernetes.NewForConfig(oc.AdminConfig())
+	if err != nil {
+		g.Fail(fmt.Sprintf("failed to create kubernetes client: %v", err))
+	}
+	r.mclient, err = pov1client.NewForConfig(oc.AdminConfig())
+	if err != nil {
+		g.Fail(fmt.Sprintf("failed to create monitoring client: %v", err))
+	}
+
+	g.BeforeEach(func() {
+		if !exutil.IsTechPreviewNoUpgrade(oc) {
+			g.Skip("skipping, this feature is only supported on TechPreviewNoUpgrade clusters")
+		}
+	})
+
+	g.AfterEach(func() {
+		_, err = r.kclient.CoreV1().ConfigMaps(operatorNamespaceName).Get(tctx, operatorConfigurationName, metav1.GetOptions{})
+		if err == nil {
+			err = r.kclient.CoreV1().ConfigMaps(operatorNamespaceName).Delete(tctx, operatorConfigurationName, metav1.DeleteOptions{})
+			if err != nil {
+				g.Fail(fmt.Sprintf("failed to delete operator configuration: %v", err))
+			}
+		}
+	})
+
+	g.Context("collection-profiles feature-set", func() {
+		g.It("should have no default profile enforcements when not enabled", func() {
+			monitors, err := r.fetchMonitorsFor( /* no profile */ )
+			o.Expect(err).To(o.BeNil())
+			o.Expect(monitors.Items).To(o.BeEmpty())
+		})
+		g.It("should have no dedicated selectors initially", func() {
+			selectors, err := r.fetchPrometheusSelectors()
+			o.Expect(err).To(o.BeNil())
+			for _, selector := range selectors {
+				for _, matchExpression := range selector.MatchExpressions {
+					o.Expect(matchExpression.Key).ToNot(o.Equal(collectionProfileFeatureLabel))
+				}
+			}
+		})
+		g.It("should have expected selectors when a profile is applied", func() {
```

Comment:
IMHO this test is too tightly coupled with the implementation.

Another option would be to check that the `profile:cluster_monitoring_operator_collection_profile:max` metric exposes the configured profile (it should be present for both minimal and full profiles). If we want to validate from an end-user perspective, I'd pick a well-known metric which is present with the full profile but not with minimal.

Reply:
Ah, agreed! I was hesitant on adding this testcase but I wasn't sure what we can do to hit the 5-test mark, but the metrics approach seems to fit perfectly here while not being too close to the implementation details, since this testcase is itself covered by CMO's test suite.